samples/html-rewrite/html-rewrite-ts/src/index.ts

```diff
@@ -0,0 +1,47 @@
+import { Kv, Variables } from '@fermyon/spin-sdk';
+import { AutoRouter, html } from 'itty-router';
+import * as cheerio from 'cheerio'
+
+const dec = new TextDecoder();
+let router = AutoRouter();
+router
+    .all("*", async (req: Request) => {
+        let url = new URL(Variables.get("upstream_url") || "");
+        url.pathname = new URL(req.url).pathname;
+        const useKvStore = (Variables.get("use_kv_store") || "false") === "true";
+        const kv = Kv.openDefault();
+
+        if (useKvStore) {
+            const original = kv.get(url.toString());
+            if (!!original) {
+                console.log('cache hit')
+                const originalHtml = dec.decode(original!)
+                return html(rewrite(originalHtml))
+            }
+        }
+        let res = await fetch(url);
+        if (res.status != 200) {
+            return res;
+        }
+        console.log(`content-type is ${res.headers.get("content-type")}`);
+        // only rewrite the response if it's an HTML response
+        if (res.headers.get("content-type")?.includes("text/html")) {
+            let originalHtml = await res.text();
+            if (useKvStore) {
+                kv.set(url.toString(), originalHtml)
```

Comment:
We probably need a manually set and check a TTL (configurable?) for the KV. Not a blocker for the sample though.

Reply:
thought about that as well... I did that for another sample. basically wrapping the HTML with an object that contains the timestamp.

Reply:
Is adding TTL support to the rust application the last item remaining for this PR?